src/empty.rs

```diff
@@ -6,113 +6,117 @@ use std::collections::HashMap;
 use std::error::Error;
 use std::sync::Arc;
 
+use adapter::BluetoothAdapter;
+
 const NOT_SUPPORTED_ERROR: &'static str = "Error! Not supported platform!";
 
 #[derive(Clone, Debug)]
-pub struct BluetoothAdapter { }
+pub struct EmptyAdapter { }
 
-impl BluetoothAdapter {
-    pub fn init() -> Result<BluetoothAdapter, Box<Error>> {
-        Ok(BluetoothAdapter::new())
+impl EmptyAdapter{
+    pub fn init() -> Result<EmptyAdapter, Box<Error>> {
+        Ok(EmptyAdapter::new())
     }
-
-    fn new() -> BluetoothAdapter {
-        BluetoothAdapter{ }
+     fn new() -> EmptyAdapter {
+        EmptyAdapter{ }
     }
 
-    pub fn get_id(&self) -> String {
-        String::new()
+     pub fn get_device_list(&self) -> Result<Vec<String>, Box<Error>> {
+        Err(Box::from(NOT_SUPPORTED_ERROR))
     }
+}
 
-    pub fn get_device_list(&self) -> Result<Vec<String>, Box<Error>> {
-        Err(Box::from(NOT_SUPPORTED_ERROR))
+impl BluetoothAdapter for EmptyAdapter {
```

Comment:
Given https://github.com/servo/devices/blob/4aee4f1adf97ebe79b63f8fe8627d0f9f7716f28/src/adapter.rs#L454-L459 I don't think we can probably revert these changes. Sorry for the confusion :( It's hard to keep track of what has been changed and what has not at this point.

Reply:
Previous comment used to contain a double negative: "I don't think we can". It has been updated.

Reply:
Oh! Do you want me to bring empty.rs to its previous implementation?

Reply:
Yes, I think it should work fine without any of those changes. There are enough moving parts in this pull request that we should reduce any unnecessary changes.